### PR TITLE
Document trials

### PR DESCRIPTION
**Pull request by malomarrec**
Note: the onboarding checlist section will point to https://github.com/sourcegraph/sourcegraph/pull/41993

## Test plan

This is docs only.

### CODE REVIEW

File: doc/cloud/index.md [hunk 1]
```diff
@@ -1,12 +1,22 @@
 # Sourcegraph Cloud (managed instances)
 
-Sourcegraph Cloud managed instances are private, dedicated Sourcegraph instances provisioned and managed by Sourcegraph.
+Sourcegraph Cloud is a single-tenant cloud (formerly known as managed instances). Cloud instances are private, dedicated Sourcegraph instances provisioned and managed by Sourcegraph.
 
-Sourcegraph provisions each instance in a completely isolated and secure cloud infrastructure. Access is restricted to only your organization through your enterprise VPN and/or SSO provider of choice.
+Sourcegraph provisions each instance in a completely isolated and secure cloud infrastructure. Access is restricted to only your organization through your SO provider of choice. Enterprise VPN is available upon request.
```
Comment (danielmarquespt) [inline doc/cloud/index.md:5]: SO? Sign on? Can we make it explicit, there's no real benefit in just using the initials.

**Review by danielmarquespt — APPROVED** [review]
Small nitpick, but otherwise nice update

File: doc/cloud/index.md [hunk 2]
```diff
@@ -1,12 +1,22 @@
 # Sourcegraph Cloud (managed instances)
 
-Sourcegraph Cloud managed instances are private, dedicated Sourcegraph instances provisioned and managed by Sourcegraph.
+Sourcegraph Cloud is a single-tenant cloud (formerly known as managed instances). Cloud instances are private, dedicated Sourcegraph instances provisioned and managed by Sourcegraph.
 
-Sourcegraph provisions each instance in a completely isolated and secure cloud infrastructure. Access is restricted to only your organization through your enterprise VPN and/or SSO provider of choice.
+Sourcegraph provisions each instance in a completely isolated and secure cloud infrastructure. Access is restricted to only your organization through your SSO provider of choice. Enterprise VPN is available upon request.
 
-[Contact us](https://about.sourcegraph.com/contact/sales) to get started with Sourcegraph Cloud.
+To get started with a cloud trial, just [signup](https://signup.sourcegraph.com) or [contact us](https://about.sourcegraph.com/contact/sales) if you have specific requirements (VPN) or either more than 500 users, more than 1,000 repos or very large monorepos.
```
Comment (rafax) [inline doc/cloud/index.md:7]: Should we say >5GB here instead of very large?

**Review by rafax — APPROVED** [review]

File: doc/cloud/index.md (hunk 2 above)
Comment (malomarrec) [reply doc/cloud/index.md:7]: Way better, good point

**Comment by malomarrec** [comment]
@ryphil I propose removing old Sourcegraph.com docs page, WDYT?

**Review by ryphil — APPROVED** [review]

**Pull request merged**